extras/static_code_analysis.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/sh
+
+# static_code_analysis.sh
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# This shell script is part of the NEST Travis CI build and test environment.
+# It performs the static code analysis and is invoked by 'build.sh'.
+# The script is also executed when running 'check_code_style.sh' for
+# a local static code analysis.
+#
+# NOTE: This shell script is tightly coupled to Python script
+#       'extras/parse_travis_log.py'. 
+#       Any changes to message numbers (MSGBLDnnnn) have effects on 
+#       the build/test-log parsing process.
+#
+
+# Command line parameters.
+RUNS_ON_TRAVIS=$1        # true or false, indicating whether the script is executed on Travis CI or runs local.
+INCREMENTAL=$2           # true or false, user needs to confirm befor checking a source file.
+FILE_NAMES=$3            # The list of files or a single file to be checked.
+NEST_VPATH=$4            # The high level NEST build path.
+VERA=$5                  # Name of the VERA++ executable.
+CPPCHECK=$6              # Name of the CPPCHECK executable.
+CLANG_FORMAT=$7          # Name of the CLANG-FORMAT executable.
+PEP8=$8                  # Name of the PEP8 executable.
+
+# PEP8 rules to ignore.
+PEP8_IGNORES="E121,E123,E126,E226,E24,E704"
+PEP8_IGNORES_EXAMPLES="${PEP8_IGNORES},E402"
+PEP8_IGNORES_TOPO_MANUAL="${PEP8_IGNORES_EXAMPLES},E265"
+
+# Print a message.
+# The format of the message depends on whether the script is executed on Travis CI or runs local.
+# print_msg "string1" "string2"
+print_msg() {
+  if $RUNS_ON_TRAVIS; then
+    echo "$1$2"
+  else
+    echo "$2"
+  fi
+}
+
+# Print version information.
+VERA_VERS=`$VERA --version`
+CPPCHECK_VERS=`$CPPCHECK --version | sed 's/^Cppcheck //'`
+CLANG_FORMAT_VERS=`$CLANG_FORMAT --version`
+PEP8_VERS=`$PEP8 --version`
+print_msg "MSGBLD0105: " "Following tools are in use:"
```

Comment:
I think it would be helpful to offset this block with version information with at least an empty line above and below on output to make it easier to recognise. May even underline the header line.